controller/api/destination/server.go

```diff
@@ -164,16 +164,23 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 		return status.Errorf(codes.InvalidArgument, "invalid authority: %s", err)
 	}
 
-	if ip := net.ParseIP(host); ip != nil {
-		// TODO handle lookup by IP address
-		log.Debug("Lookup of IP addresses is not supported for profiles")
-		return status.Errorf(codes.InvalidArgument, "cannot discover profiles for IP addresses")
-	}
+	var service watcher.ServiceID
+	var path string
 
-	service, _, err := parseK8sServiceName(host, s.clusterDomain)
-	if err != nil {
-		log.Debugf("Invalid service %s", dest.GetPath())
-		return status.Errorf(codes.InvalidArgument, "invalid service: %s", err)
+	if ip := net.ParseIP(host); ip != nil {
+		svc, exists := s.ips.GetSvc(ip.String())
+		if exists {
+			path = fmt.Sprintf("%s.%s.svc.%s", svc.Name, svc.Namespace, s.clusterDomain)
+		} else {
+			return status.Errorf(codes.InvalidArgument, "IP address %s is not a service", ip.String())
```

Comment:
InvalidArgument feels a bit weird, but I'm not sure what a better error would be.

It's worth noting that this works differently than looking up a service that doesn't exist by authority.  In that case we immediately send an update with the default service profile and then we'll keep the stream open and if a service profile with that name ever gets created we'll send an update.

In this case, however, if we lookup an IP that doesn't correspond to a service, we immediately fail the stream instead of sending back the default service profile at all.

@olix0r: is that how you would expect this to work from the proxy's perspective?  Would it be preferable to send the default service profile and then leave the stream open, never sending any further updates on it?

Reply:
@adleong I change it so that when an IP does not map to a service, a profile stream is still returned that receives one update, the default profile. This stream will not receive any more updates since we do not have a service to subscribe to. This does match more with what Oliver mentioned in the original issue, so I think that is what he would lean towards anyways.

Reply:
I also updated the PR description with this new behavior and added an example of testing it.